lib/rules/no-obj-calls.js

```diff
@@ -54,7 +67,7 @@ module.exports = {
                 }
 
                 for (const { node } of tracker.iterateGlobalReferences(traceMap)) {
-                    context.report({ node, messageId: "unexpectedCall", data: { name: node.callee.name } });
+                    context.report({ node, messageId: "unexpectedCall", data: { name: getReportNodeName(node) } });
```

Comment:
Thanks for fixing this bug!

Current behavior ([demo](https://eslint.org/demo#eyJ0ZXh0IjoiLyogZXNsaW50IG5vLW9iai1jYWxsczogXCJlcnJvclwiKi9cbi8qIGVzbGludC1lbnYgYnJvd3NlciovXG5cbndpbmRvdy5KU09OKCk7IC8vICd1bmRlZmluZWQnIGlzIG5vdCBhIGZ1bmN0aW9uIChuby1vYmotY2FsbHMpIiwib3B0aW9ucyI6eyJwYXJzZXJPcHRpb25zIjp7ImVjbWFWZXJzaW9uIjo2LCJzb3VyY2VUeXBlIjoic2NyaXB0IiwiZWNtYUZlYXR1cmVzIjp7fX0sInJ1bGVzIjp7fSwiZW52Ijp7ImJyb3dzZXIiOmZhbHNlfX19)):

```js
/* eslint no-obj-calls: "error"*/
/* eslint-env browser*/

window.JSON(); // 'undefined' is not a function (no-obj-calls)
```

Reply:
A question, maybe we could use `path` from [iterateglobalreferences](https://mysticatea.github.io/eslint-utils/api/scope-utils.html#tracker-iterateglobalreferences)?

I guess it depends on whether we want to report "JSON is not a function" or "foo is not a function" in cases such as the following:

```js
/* eslint no-obj-calls: "error"*/

var foo = bar ? baz : JSON;

foo();
```

Reply:
Ok, I'll try it! :)

Reply:
I'm not sure, let's wait for more opinions.

Current behavior reports "foo is not a function". 

I'd vote for "JSON is not a function" as it might be a bit more helpful for the user to find out what's actually happening and why is the location of "foo()" reported as a global object call.

Reply:
> I'm not sure, let's wait for more opinions.  

Ok. And just to note. 
The reason for my preference was that `foo is not a function` looks like a more intuitive explanation for the code. 

> I'd vote for "JSON is not a function" as it might be a bit more helpful for the user to find out what's actually happening and why is the location of "foo()" reported as a global object call.

But I agree that `JSON is not a function` is more helpful to find out what's actually happening

Reply:
Could we do both? "foo is a reference to JSON, which is not a function"? That seems like it gives the right amount of information without being too confusing or requiring you to figure out what the code is doing.

Reply:
> Could we do both? "foo is a reference to JSON, which is not a function"?

That would be ideal!

I'd be also fine with "foo is not a function" in this PR (it's already much better than "undefined is not a function") and improving the message further in another PR.

Reply:
> Could we do both? "foo is a reference to JSON, which is not a function"? 


Agree! it would make the message more clear. maybe it needs new messageId like -`unexpectedRefCall`. I will do it. thanks :)